pywbem/_subscription_manager.py

```diff
@@ -344,29 +346,38 @@ def add_server(self, server):
         self._owned_filters[server_id] = []
         self._owned_destinations[server_id] = []
 
-        # Recover any owned destination, filter, and subscription instances
-        # that exist on this server
+        # Get the SystemName from the WBEMServer CIM_CimomManager for this
+        # server.
+        self._systemnames[server_id] = server.cimom_inst['SystemName']
 
-        this_host = getfqdn()
+        # Issue #2709 Expand to account for WBEMServer profile
 
+        # Get the hostname of the client system to be part of destination Name
+        this_client = getfqdn()
+        svr_system_name = self._systemnames[server_id]
+
+        # Recover owned destination, filter, and subscription instances
+        # that exist on the WBEMServer
         dest_name_pattern = re.compile(
             _format(r'^pywbemdestination:owned:{0}:{1}:[^:]*$',
-                    this_host, self._subscription_manager_id))
+                    this_client, self._subscription_manager_id))
         dest_insts = server.conn.EnumerateInstances(
             DESTINATION_CLASSNAME, namespace=server.interop_ns)
+
         for inst in dest_insts:
             if re.match(dest_name_pattern, inst.path.keybindings['Name']) \
-                    and inst.path.keybindings['SystemName'] == this_host:
+                    and inst.path.keybindings['SystemName'] == svr_system_name:
```

Comment:
We should remove the SystemName comparison from this condition, since the identification can now completely be done through the Name property.

Also, for providers that set the SystemName key, we cannot be sure whether the SystemName key in CIM_ObjectManager is consistent with the SystemName key in CIM_ListenerDetination and CIM_IndicationFilter, so the SystemName comparison might even not work.

This comment also applies to `create_filter()`.

Reply:
DONE